pkg/controller/controller.go

```diff
@@ -110,26 +110,45 @@ type controller struct {
 	brokerLister           listers.BrokerLister
 	serviceClassLister     listers.ServiceClassLister
 	instanceLister         listers.InstanceLister
+	bindingLister          listers.BindingLister
+	brokerQueue            workqueue.RateLimitingInterface
+	serviceClassQueue      workqueue.RateLimitingInterface
+	instanceQueue          workqueue.RateLimitingInterface
+	bindingQueue           workqueue.RateLimitingInterface
 }
 
 // Run runs the controller until the given stop channel can be read from.
-func (c *controller) Run(stopCh <-chan struct{}) {
+func (c *controller) Run(workers int, stopCh <-chan struct{}) {
 	defer runtimeutil.HandleCrash()
+	defer func() {
+		c.brokerQueue.ShutDown()
+		c.serviceClassQueue.ShutDown()
+		c.instanceQueue.ShutDown()
+		c.bindingQueue.ShutDown()
+	}()
```

Comment:
This implicitly depends on `<-stopCh` below, which is the overall signal to shut everything down. Is there a way to explicitly depend on it? Or maybe a comment somehow? I think I just need to read it again a couple more times. There's a couple of moving pieces, but nothing looks wrong to me.